# 1. Introduction
__Global Energy Consumption__ has increased rapidly over the past few years due to industrilization, population growth, urban developement, and technological advancement. The Excessive dependence on fossil fuels such as coal, oil, and natural gas has led to serious environmental challenges including climate change, global warming, and increasig carbon dioxide (CO<sub>2</sub>) emissions.

The dataset provides __five years (2020-2024)__ of data combining __global energy consumption__ statistics for __50__ countries across all continents.

### The project focuses on:
- Renewable vs non-renewable energy usage
- CO₂ emissions trends
- GDP and population relationship with energy usage
- Country-wise energy efficiency
- Future sustainability insights
- SQL integration for analytical queries
- Predictive analysis using Machine Learning

By analyzing these factors, the project provides actionable insights and recommendations that can help in promoting renewable energy adoption, improving energy efficiency, and supporting sustainable developement goals.

---
# 2. Tools & Technologies Used
- Python(pandas, Numpy)
- Data Visualization(Matplotlib, Seaborn, Plotly)
- Database(MySQL/SQLite)
- ML (Scikit-learn)
- Dashboard(Streamlit)
- Notebook(Jupyter)
---

### Dataset Source
- [Climate & Energy Consumption Dataset 2020-2024](https://www.kaggle.com/datasets/emirhanakku/climate-and-energy-consumption-dataset-20202024)

# 3. Import Libraries

In [ ]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import plotly.express as px
import sqlite3

from sklearn.linear_model import LinearRegression
from sklearn.model_selection import train_test_split
from sklearn.metrics import mean_absolute_error, r2_score

import warnings
warnings.filterwarnings('ignore')

# 4. Loading Data

In [2]:
df = pd.read_csv("../Dataset/global_climate_energy.csv", lineterminator="\n")

In [12]:
df.head()

,date,country,avg_temperature,humidity,co2_emission,energy_consumption,renewable_share,urban_population,industrial_activity_index,energy_price
0,2020-01-01,Germany,28.29,31.08,212.63,11348.75,14.42,76.39,51.22,83.93
1,2020-01-02,Germany,28.38,37.94,606.05,4166.64,5.63,86.26,78.27,110.40
2,2020-01-03,Germany,28.74,57.67,268.72,4503.80,14.20,75.92,48.96,173.58
3,2020-01-04,Germany,26.66,51.34,167.32,3259.13,13.84,63.15,97.42,89.13
4,2020-01-05,Germany,26.81,65.38,393.89,7023.72,6.93,76.02,81.89,40.60


In [15]:
df.tail()

,date,country,avg_temperature,humidity,co2_emission,energy_consumption,renewable_share,urban_population,industrial_activity_index,energy_price
36535,2024-12-27,Indonesia,10.91,85.88,134.23,6607.39,22.04,69.43,76.38,160.33
36536,2024-12-28,Indonesia,10.91,65.14,366.68,4654.41,23.79,62.66,43.33,58.49
36537,2024-12-29,Indonesia,10.91,66.33,256.12,1982.89,14.92,77.53,76.67,148.77
36538,2024-12-30,Indonesia,10.91,39.18,251.60,2689.54,21.66,62.92,72.84,153.20
36539,2024-12-31,Indonesia,10.91,54.24,444.88,6227.33,22.28,73.13,66.80,185.53


In [13]:
df.shape

(36540, 10)

In [14]:
df.columns

Index(['date', 'country', 'avg_temperature', 'humidity', 'co2_emission',
       'energy_consumption', 'renewable_share', 'urban_population',
       'industrial_activity_index', 'energy_price'],
      dtype='object')

In [16]:
df.dtypes

date                          object
country                       object
avg_temperature              float64
humidity                     float64
co2_emission                 float64
energy_consumption           float64
renewable_share              float64
urban_population             float64
industrial_activity_index    float64
energy_price                 float64
dtype: object

The dataset contains 36540 rows of 50 unique countries and have 10 columns:
- date
- country name
- average temperature
- humidity
- CO<sub>2</sub> emision
- energy consumption
- renewable share
- urban population
- industrial activity index
- energy price


# 5. Data Cleaning

### 5.1 Check Missing value

In [18]:
df.isnull().sum()

date                         0
country                      0
avg_temperature              0
humidity                     0
co2_emission                 0
energy_consumption           0
renewable_share              0
urban_population             0
industrial_activity_index    0
energy_price                 0
dtype: int64

### 5.2 Check Duplicate Rows

In [20]:
print(df.duplicated().sum())

0


__Data is already clean and niether have missing values nor duplicate rows__

# 6. Feature Engineering

### 6.1 Renewable Percentage Column

In [22]:
df["renewable_percentage"] = ((df["renewable_share"] / df["energy_consumption"]) * 100)

### 6.2 Energy Per Capita

In [23]:
df["energy_per_capita"] = (df["energy_consumption"] / df["urban_population"])

### 6.3 CO<sub>2</sub> Per Capita

In [24]:
df["co2_per_capita"] = (df["co2_emission"] / df["urban_population"])